benchmark/syclblas/extension/omatcopy_batched.cpp

```diff
@@ -0,0 +1,184 @@
+  {
+    auto m_b_temp_gpu =
+        blas::make_sycl_iterator_buffer<scalar_t>(m_b_temp, size_b);
+
+    auto event = blas::_omatcopy_batch(sb_handle, *t_str, m, n, alpha, m_a_gpu,
+                                       lda, stride_a, m_b_temp_gpu, ldb,
+                                       stride_b, batch_size);
+
+    sb_handle.wait();
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(m_b_temp, m_b_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_warmup_method_def = [&]() -> void {
+    auto event =
+        blas::_omatcopy_batch(sb_handle, *t_str, m, n, alpha, m_a_gpu, lda,
+                              stride_a, m_b_gpu, ldb, stride_b, batch_size);
+    return;
+  };
+
+  auto blas_method_def = [&]() -> std::vector<cl::sycl::event> {
+    auto event =
+        blas::_omatcopy_batch(sb_handle, *t_str, m, n, alpha, m_a_gpu, lda,
+                              stride_a, m_b_gpu, ldb, stride_b, batch_size);
+    sb_handle.wait(event);
+    return event;
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_warmup_method_def);
+  sb_handle.wait();
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  state.SetItemsProcessed(state.iterations() * state.counters["n_fl_ops"]);
+  state.SetBytesProcessed(state.iterations() *
+                          state.counters["bytes_processed"]);
+
+  blas_benchmark::utils::calc_avg_counters(state);
+}
+
+template <typename scalar_t>
+void register_benchmark(blas_benchmark::Args& args,
+                        blas::SB_Handle* sb_handle_ptr, bool* success) {
+  auto omatcopy_batch_params =
+      blas_benchmark::utils::get_matcopy_batch_params<scalar_t>(args);
+
+  for (auto p : omatcopy_batch_params) {
+    std::string ts;
+    index_t m, n, lda_mul, ldb_mul, stride_a_mul, stride_b_mul, batch_size;
+    scalar_t alpha;
+    std::tie(ts, m, n, alpha, lda_mul, ldb_mul, stride_a_mul, stride_b_mul,
+             batch_size) = p;
+    int t = static_cast<int>(blas_benchmark::utils::to_transpose_enum(ts));
+
+    auto BM_lambda = [&](benchmark::State& st, blas::SB_Handle* sb_handle_ptr,
+                         int t, index_t m, index_t n, scalar_t alpha,
+                         index_t lda_mul, index_t ldb_mul, index_t stride_a_mul,
+                         index_t stride_b_mul, index_t batch_size,
+                         bool* success) {
+      run<scalar_t>(st, sb_handle_ptr, t, m, n, alpha, lda_mul, ldb_mul,
+                    stride_a_mul, stride_b_mul, batch_size, success);
+    };
+    benchmark::RegisterBenchmark(
+        get_name<scalar_t>(ts, m, n, alpha, lda_mul, ldb_mul, stride_a_mul,
```

Comment:
Can we use the `get_name` function inside the `blas_benchmark::utils::` namespace instead of this local definition of `get_name`? Same goes for the extension operators that have been merged already.